ksql-version-metrics-client/src/test/java/io/confluent/ksql/version/metrics/collector/BasicCollectorTest.java

```diff
@@ -16,85 +16,97 @@
 import io.confluent.support.metrics.common.Version;
-import io.confluent.support.metrics.common.time.Clock;
-import io.confluent.support.metrics.common.time.TimeUtils;
-import java.util.Collection;
-import java.util.EnumSet;
-import org.hamcrest.CoreMatchers;
-import org.junit.Assert;
+import java.time.Clock;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
-@RunWith(Parameterized.class)
+@RunWith(MockitoJUnitRunner.class)
 public class BasicCollectorTest {
 
-  public static class MockClock implements Clock {
-    private long currentTime = 0;
+  private static final KsqlModuleType MODULE_TYPE = KsqlModuleType.SERVER;
 
-    public MockClock() {
-    }
+  private BasicCollector basicCollector;
+  @Mock
+  private Clock clock;
+  @Mock
+  private Supplier<Boolean> activenessStatusSupplier;
 
-    public long currentTimeMs() {
-      return currentTime;
-    }
-
-    public void setCurrentTimeMillis(final long timeMillis) {
-      currentTime = timeMillis;
-    }
+  @Before
+  public void setUp() throws Exception {
+    when(clock.millis()).thenReturn(12345L);
+    when(activenessStatusSupplier.get()).thenReturn(true);
+    basicCollector = new BasicCollector(MODULE_TYPE, activenessStatusSupplier, clock);
   }
 
-  @Parameterized.Parameters
-  public static Collection<KsqlModuleType> data() {
-    return EnumSet.allOf(KsqlModuleType.class);
-  }
+  @Test
+  public void testCollectMetricsAssignsCurrentTime() {
+    // Given:
+    final long t1 = 1000L;
+    final long t2 = 1001L;
+    when(clock.millis())
+        .thenReturn(TimeUnit.SECONDS.toMillis(t1))
+        .thenReturn(TimeUnit.SECONDS.toMillis(t2));
 
-  @Parameterized.Parameter
-  public KsqlModuleType moduleType;
+    // When:
+    KsqlVersionMetrics metrics = basicCollector.collectMetrics();
 
-  private MockClock mockClock;
-  private TimeUtils timeUtils;
+    // Then:
+    assertThat(metrics.getTimestamp(), is(t1));
 
-  @Before
-  public void setUp() throws Exception {
-    mockClock = new MockClock();
-    timeUtils = new TimeUtils(mockClock);
+    metrics = basicCollector.collectMetrics();
+    assertThat(metrics.getTimestamp(), is(t2));
   }
 
   @Test
-  public void testGetCollector() {
-    final BasicCollector basicCollector = new BasicCollector(moduleType, timeUtils);
+  public void shouldSetActivenessToTrue() {
+    // Given:
+    when(activenessStatusSupplier.get()).thenReturn(true);
 
-    final KsqlVersionMetrics expectedMetrics = new KsqlVersionMetrics();
-    expectedMetrics.setTimestamp(timeUtils.nowInUnixTime());
-    expectedMetrics.setConfluentPlatformVersion(Version.getVersion());
-    expectedMetrics.setKsqlComponentType(moduleType.name());
+    // When:
+    final KsqlVersionMetrics merics = basicCollector.collectMetrics();
```

Comment:
```suggestion
    final KsqlVersionMetrics metrics = basicCollector.collectMetrics();
```
typo appears in the tests below as well